core/templates/dev/head/domain/exploration/QuestionObjectFactory.js

```diff
@@ -79,6 +80,18 @@ oppia.factory('QuestionObjectFactory', [function() {
     return this._answerGroups.length > 0;
   };
 
+  Question.prototype.getId = function () {
+    return this.getStateName().toLowerCase().split(' ').join('');
```

Comment:
Used ':' and '_' as delimiters, like this: 'sidebaritem:Question_1:bridgetext' .